seep-worker/src/main/java/uk/ac/imperial/lsds/seepworker/core/output/FileOutputBuffer.java

```diff
@@ -35,17 +43,50 @@ public FileOutputBuffer(DataReference dr, int batchSize) {
 	private BufferedOutputStream createOutputFile(int batchSize) {
 		String path = dr.getDataStore().getConfig().getProperty(FileConfig.FILE_PATH);
 		String pathAndFilename = path + id;
-		Path p = FileSystems.getDefault().getPath(pathAndFilename);
-		File outputFile = p.toFile();
-		BufferedOutputStream bws = null;
-		try {
-			// Configured with the given batch size
-			bws = new BufferedOutputStream(new FileOutputStream(outputFile), batchSize);
-		} catch (FileNotFoundException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+		Boolean isHDFS = new Boolean(dr.getDataStore().getConfig().getProperty(FileConfig.HDFS_SOURCE));
+		if (isHDFS) {
+			String hdfsUri = dr.getDataStore().getConfig().getProperty(FileConfig.HDFS_URI);
+			//We have two Path types in this file, and the other is imported, so
+			//fully qualify this one.
+			org.apache.hadoop.fs.Path hdfsPath = new org.apache.hadoop.fs.Path(hdfsUri + pathAndFilename);
+			try {
+				FileSystem fs = FileSystem.get(hdfsPath.toUri(), new Configuration());
+				FSDataOutputStream hdfsOutput;
+				try {
+					hdfsOutput = fs.create(hdfsPath, false);
+				} catch (IOException ioe) {
```

Comment:
Another commono exception here is not having the permission to create the file (Since HDFS has users and permissions) We might need to handle that case separatly and print a message to the user..